### ISSUE TITLE
Allow to update cloud config after bootstrap

### ISSUE THREAD

**Issue by mudler** [rancher-sandbox/os2#120]
Currently the cloud config specified to bootstrap a node is consumed only during installation, not giving any way to update it on the machine.

This card is about having a mechanism that reacts to changes to the cloudConfig and propagates the updates over the nodes. This can be done in different ways, via specific workloads or for example, we could try to re-use system-upgrade-controller for such task.

**Comment by kkaempf** [comment]
I'd vote against this issue, as it will bring confusion and chaos. Like people expecting to re-partition (without losing data 😆 )

In the container world, systems are considered 'cattle'. If one doesn't behave, you shoot it down and re-install. That's what we should offer for this use-case.

**Comment by mudler** [comment]
yes, I find it as well confusing. I'd more thorn to a solution among the lines of https://github.com/rancher-sandbox/cOS-toolkit/issues/749, which is completely decoupled from the current implementation, and "opt-in"

**Comment by mudler** [comment]
I think this can be already achieved with system-upgrade-controller by the way. It just needs to be instrumented specifically to do so.
